libhpcalcs/src/prime_cmd.c

```diff
@@ -360,47 +364,124 @@ HPEXPORT int HPCALL calc_prime_r_recv_screen(calc_handle * handle, calc_screensh
     return res;
 }
 
-// Seems to be made of a series of CMD_PRIME_RECV_FILE.
+HPEXPORT int HPCALL calc_prime_s_enable_new_protocol(calc_handle * handle) {
+    int res;
+    res = prime_send_new_protocol_init(handle);
+    if (!res) {
+        handle->protocol_version = 1;
+    }
+    return res;
+}
+
+HPEXPORT int HPCALL calc_prime_r_enable_new_protocol(calc_handle * handle) {
+    int res = 0;
+    if (handle == NULL) {
+        res = ERR_INVALID_HANDLE;
+        hpcalcs_error("%s: handle is NULL", __FUNCTION__);
+    }
+    return res;
+}
+
+HPEXPORT int HPCALL calc_prime_s_disable_new_protocol(calc_handle * handle) {
+    int res;
+    // It looks like sending a status check using the old protocol drops the Prime
+    // out of the new protocol mode.
+    res = calc_prime_s_check_ready(handle);
+    if (res) {
+        return res;
+    }
+    return res;
+}
+
+HPEXPORT int HPCALL calc_prime_r_disable_new_protocol(calc_handle * handle) {
+    int res;
+    if (handle != NULL) {
+        // While the prime is in thew new protocol mode, it sends us packages
+        // starting with 0xFE. We need to eliminate those now (and also wait for
+        // the status response).
+        res = calc_prime_r_check_ready(handle, NULL, NULL);
+        handle->protocol_version = 0;
+    }
+    else {
+        res = ERR_INVALID_HANDLE;
+        hpcalcs_error("%s: handle is NULL", __FUNCTION__);
+    }
+    return res;
+}
+
 HPEXPORT int HPCALL calc_prime_s_send_file(calc_handle * handle, files_var_entry * file) {
     int res;
     if (handle != NULL && file != NULL) {
+        uint32_t offset = 0;
+        uint32_t header_size = 8;
         uint8_t namelen = (uint8_t)char16_strlen(file->name) * 2;
-        uint32_t size = 10 - 6 + namelen + file->size; // Size of the data after the header.
-        prime_vtl_pkt * pkt = prime_vtl_pkt_new(size + 6); // Add size of the header.
-        hpcalcs_debug("Virtual packet has size %" PRIu32 " (%" PRIx32 ")\n", size, size);
+        uint32_t size = namelen + file->size + 12; // Size of the data plus something
```

Comment:
No idea why there are two slightly different file sizes. They never seem to diverge by anything other than a constant (even when the connectivity kit switches to compressed transfers).